tests/deployment_test.go

```diff
@@ -234,53 +235,31 @@ func (d *DeploymentSuite) TestGetCurrentDeployment_NoCurrentDeployment() {
 	}
 }
 
-// addDeploymentsAndVerifyListDeployments does the following:
-// - starts deployment workflow(s)
-// - makes a ListDeployment request (with and without seriesFilter)
-// - verifies that all the expected deployments eventually appear in the list.
-func (d *DeploymentSuite) addDeploymentsAndVerifyListDeployments(seriesName string, buildID string, withSeriesFilter bool) {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
-	defer cancel()
-
-	// todo (shivam) - fails when seriesName has backslashes
-	taskQueue := &taskqueuepb.TaskQueue{Name: "deployment-test", Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
-	workerDeployment := &deploymentpb.Deployment{
-		SeriesName: seriesName,
-		BuildId:    buildID,
+// verifyDeploymentListInfo checks the equality between two DeploymentListInfo objects
+func (d *DeploymentSuite) verifyDeploymentListInfo(expectedDeploymentListInfo *deploymentpb.DeploymentListInfo, receivedDeploymentListInfo *deploymentpb.DeploymentListInfo) bool {
+	maxDurationBetweenTimeStamps := 1 * time.Millisecond
+	if expectedDeploymentListInfo.Deployment.SeriesName != receivedDeploymentListInfo.Deployment.SeriesName {
+		return false
 	}
-
-	// Starting a deployment workflow
-	errChan := make(chan error)
-	defer close(errChan)
-
-	// Starting a deployment workflow
-	go func() {
-		d.startDeploymentWorkflows(ctx, taskQueue, workerDeployment, errChan)
-	}()
-
-	request := &workflowservice.ListDeploymentsRequest{
-		Namespace: d.Namespace(),
+	if expectedDeploymentListInfo.Deployment.BuildId != receivedDeploymentListInfo.Deployment.BuildId {
+		return false
 	}
-
-	if withSeriesFilter {
-		request.SeriesName = seriesName
-		d.validateListDeploymentsResponse_WithSeriesFilter(ctx, request, workerDeployment)
-	} else {
-		d.validateListDeploymentsResponse_WithoutSeriesFilter(ctx, request)
+	if expectedDeploymentListInfo.IsCurrent != receivedDeploymentListInfo.IsCurrent {
+		return false
 	}
-
-	<-ctx.Done()
-	select {
-	case err := <-errChan:
-		d.Fail("Expected error channel to be empty but got error %w", err)
-	default:
+	if expectedDeploymentListInfo.CreateTime.AsTime().Sub(receivedDeploymentListInfo.CreateTime.AsTime()) > maxDurationBetweenTimeStamps {
+		return false
 	}
+	return true
 }
 
-func (d *DeploymentSuite) validateListDeploymentsResponse_WithSeriesFilter(ctx context.Context,
-	request *workflowservice.ListDeploymentsRequest, expectedDeployment *deploymentpb.Deployment) {
-	expectedNumberOfDeployments := 1
+// verifyDeployments does the following:
+// - makes a list deployments call with/without seriesFilter
+// - validates the response with expectedDeployments
+func (d *DeploymentSuite) verifyDeployments(ctx context.Context, request *workflowservice.ListDeploymentsRequest,
```

Comment:
Do you need pass the request if you are passing the series filter?

Reply:
yes, because I am making an API call here - moreover, I intend for this function to do both the api call and the validation and further splitting that doesn't make sense as it makes harder for a new reader to read with so much splitting happening

wdyt?

Reply:
If `seriesFilter` and `request.SeriesName` are going to have the same value whenever your function is called (which is the case) then your function should not get both, otherwise it would be confusing and the reader would wonder why you get the same info twice.